.github/workflows/dart_services.yml

```diff
@@ -29,25 +29,31 @@ jobs:
     steps:
       - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744
       - uses: dart-lang/setup-dart@d6a63dab3335f427404425de0fbfed4686d93c4f
-        with:
-          sdk: stable
       - name: Install dependencies
         run: sudo apt-get install -y protobuf-compiler redis
       - name: Install dependencies
         run: dart pub get
       - name: Analyze code
         run: dart analyze --fatal-infos .
+      - name: check formatting
+        run: dart format --set-exit-if-changed lib test tool  
       - name: Prepare Flutter
         run: |
           dart run tool/update_sdk.dart ${{ matrix.sdk }}
           export PATH=$PATH:$PWD/flutter-sdks/${{ matrix.sdk }}/bin
           flutter doctor -v
           flutter config --enable-web
       - name: Run tests
+        env:
+          FLUTTER_CHANNEL: ${{ matrix.sdk }}
         run: |
           export PATH=$PATH:$HOME/.pub-cache/bin
           dart pub global activate protoc_plugin
-          ./tool/ci.sh ${{ matrix.sdk }}
+          # Prepare to run unit tests (but do not actually run tests).
+          dart pub run grinder buildbot
+          # Ensure that we've uploaded the compilation artifacts to google storage.
+          dart pub run grinder validate-storage-artifacts
```

Comment:
```suggestion
          dart run grinder buildbot
          # Ensure that we've uploaded the compilation artifacts to google storage.
          dart run grinder validate-storage-artifacts
```

Reply:
Ah, I missed that this was a commit suggestion.